### PR TITLE
DRILL-7999 Base Docker image on maven:3.8.2-jdk-11 and openjdk-11

### PR DESCRIPTION
**Pull request by dzamo**
# [DRILL-7999](https://issues.apache.org/jira/browse/DRILL-7999): Base Docker image on maven:3.8.2-jdk-11 and openjdk-11

## Description

Our Docker container builds on OpenJDK 8u232 (released late in 2019), while the current LTS series of OpenJDK is 11.  Update our Dockerfile to OpenJDK 11.

## Documentation

Unchanged except that I do address the following docs issues pointed out by a user, not related to this JDK version change.

> 1. Drill documentation referrs to $DRILL_HOME environment variable which is not set in the Docker Image. Additionally in the doucmenation says it can be found in /etc/drill/conf while in the Docker image Drill has been installed into /opt/drill/conf
> 2. Add information how to add (mount) custom configuration files into the Docker image or even better, just make sure there is a Docker section in Drill's documentation whereever appropriate

## Testing
- Test Drill built from master branch (1.20-SNAPSHOT) with OpenJDK 11 outside of Docker.
- Test Drill built from master branch (1.20-SNAPSHOT) with OpenJDK 11 inside of Docker by building a Docker image from the updated Dockerfile.
- Test a downloaded build of Drill 1.19 inside a Docker container based on OpenJDK 11 (published [here](https://hub.docker.com/repository/docker/dzamo2/drill)).

### CODE REVIEW

**Review by luocooong — APPROVED** [review]
LGTM. +1

**Comment by dzamo** [comment]
@vvysotskyi had a great idea of turning the base openjdk image name in the Dockerfile into a variable so that we can build and publish for a few different JDK versions with this one Dockerfile. So please don't merge this until I return with that enhancement, thanks.

**Comment by luocooong** [comment]
@dzamo If possible, rebase the code on the master. It included a Travis CI bugfix.

File: Dockerfile
Comment (vvysotskyi) [inline Dockerfile:26]: Are you sure that jars built using JDK 11 will be running fine on JDB 8?

File: Dockerfile
Comment (vvysotskyi) [inline Dockerfile:57]: Please assign a default value for `BASE_IMAGE` to preserve backward compatibility when building images without specifying env variables.

**Review by vvysotskyi — CHANGES_REQUESTED** [review]
@dzamo, could you please also try setting up the build on DockerHub for your fork and share here instructions of what should be updated there to have multiple builds?

File: Dockerfile
Comment (dzamo) [reply Dockerfile:26]: Oh I better check this, thanks

Comment (dzamo) [reply Dockerfile:26]: Reverted the build container to OpenJDK 8

**Comment by dzamo** [comment]
@vvysotskyi I'm just stuck with the DockerHub builds.  In my personal account the Builds tab shows

> Available on Pro and Team plans.

File: Dockerfile
Comment (vvysotskyi) [inline Dockerfile:29]: It can be made configurable too.

File: Dockerfile
Comment (vvysotskyi) [inline Dockerfile:25]: Please use JDK 8 as a default one.

**Comment by vvysotskyi** [comment]
@dzamo, ok, thanks for the clarification, I didn't expect that it is not available for a free account.

Regarding the latest, let's leave JDK 8 as it is now.

File: hooks/build
Comment (vvysotskyi) [inline hooks/build:3]: We don't support JDK 17 yet: https://github.com/apache/drill/blob/master/.github/workflows/ci.yml#L36

**Comment by dzamo** [comment]
@luocooong I rebased on master but I see my latest Travis CI ARM64 build failed.  Most informative line I saw in the logs

> [ERROR] Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "WorkManager.StatusThread"

Is it something we already know of?  I am not very knowledgeable about tweaking the build but if there's anything I can help with let me know...

cc @vdiravka

**Review by vvysotskyi — APPROVED** [review]
@dzamo, thanks for making changes, looks good, +1.
Could you please push the branch as we discussed to test that it would work as we expect before merging it?

**Comment by vvysotskyi** [comment]
Also, please squash commits and update the commit message to include the ticket number.

**Review by vvysotskyi — CHANGES_REQUESTED** [review]
It fails on DockerHub:

> Executing build hook...
> Could not execute hook at 'hooks/build'. Is it missing a #! line?

**Comment by dzamo** [comment]
> It fails on DockerHub:
> 
> > Executing build hook...
> > Could not execute hook at 'hooks/build'. Is it missing a #! line?

Added the shebang.  And I checked that the two hooks have the executable perm set for all users.

File: hooks/build
Comment (vvysotskyi) [inline hooks/build:22]: Just recalled the latest tag is not published for the master version. It is published for release only, so please delete it here. We use the `master` tag for builds from the master branch. The good news is that according to https://docs.docker.com/docker-hub/builds/advanced/#environment-variables-for-building-and-testing, we can use tag name from the dockerhub configuration: `DOCKER_TAG`, so please replace `latest` with that value, and add this variable to tag names with open-jdk, so we will have something like `$DOCKER_TAG-openjdk:8-jre` that will become `master-openjdk:8-jre` or `latest-openjdk:8-jre` or `1.20.0-openjdk:8-jre` for the release.

File: hooks/push
Comment (vvysotskyi) [inline hooks/push:20]: Please also use `DOCKER_TAG` var here, and add one more line to push `apache/drill:$DOCKER_TAG` (tag without openjdk suffix)

**Comment by cgivre** [comment]
@dzamo I have a silly request for DockerHub as well... If this is the "official" Drill container, could you see if you could add the logo to DockerHub?

**Comment by vvysotskyi** [comment]
@cgivre, we have investigated it some time ago and it cannot be done, please see this ticket: https://issues.apache.org/jira/browse/INFRA-19810
But we can add a badge for DockerHub into the root README.md: https://github.com/badges/shields

**Comment by dzamo** [comment]
> @dzamo I have a silly request for DockerHub as well... If this is the "official" Drill container, could you see if you could add the logo to DockerHub?

@cgivre for sure.  I'll ask @vvysotskyi for help here since I think he's the only one of us with access.  We should probably get at least one more of us access there, maybe ideally one in a distinct time zone.  Is it more of thing that the PMC members cover?  I don't mind putting my hand up if not, and then I can sort out the logo once I'm added.

**Comment by vvysotskyi** [comment]
@dzamo, see my comment above.
Regarding access to DockerHub repo, I think committers also can obtain it by creating a ticket to infra, specifying the username and reason for obtaining access.

**Comment by dzamo** [comment]
I finally got to the bottom of
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.2.0:single (distro-assembly) on project distribution: Failed to create assembly: Error adding file to archive: /src/distribution/../git.properties -> [Help 1]
```
It came from adding .git/ to .dockerignore.  Turns out one can't build without the git repository there because of the git-commit-id plugin.  Here's hoping some new image tags will show up in Docker Hub in the next couple of hours.

**Review by vvysotskyi — APPROVED** [review]
@dzamo, thanks for making changes, now the build passed and the required tags were published, so it can be merged.

**Pull request merged**
